### PR TITLE
Add example source for C# and VB to roundtrip JSON de/serialize

### LINKED ISSUE #33318: Add `DateOnly` and `TimeOnly` examples in the STJ content
**Issue description**

Add to the existing `System.Text.Json` serialization content, details and examples about serializing and deserializng `DateOnly` and `TimeOnly` properties, which weren't originally supported prior to .NET 7.

https://learn.microsoft.com/dotnet/standard/serialization/system-text-json/how-to?pivots=dotnet-7-0

Related to https://github.com/dotnet/runtime/issues/53539 and https://github.com/dotnet/runtime/pull/69160

---
[Associated WorkItem - 58104](https://dev.azure.com/msft-skilling/Content/_workitems/edit/58104)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Add example source for C# and VB to roundtrip JSON de/serialize

Fixes #33318

### CODE REVIEW

File: docs/standard/serialization/system-text-json/how-to.md [hunk 1]
```diff
@@ -85,6 +85,18 @@ The preceding examples use type inference for the type being serialized. An over
...
+:::code source="snippets/system-text-json-how-to/cs/RoundtripDateOnly.cs" id="SeriesDataPoint":::
+:::code source="snippets/system-text-json-how-to/vb/RoundtripDateOnly.vb" id="SeriesDataPoint":::
+
+The following example serializes the `SeriesDataPoint` object, displays the resulting JSON, and then deserializes it back into a new instance of the `SeriesDataPoint` type. Finally, the original and newly deserialized instances are compared for equality and the results are written to the console:
+
+:::code source="snippets/system-text-json-how-to/cs/RoundtripDateOnly.cs" id="RoundtripDateOnly":::
+:::code source="snippets/system-text-json-how-to/vb/RoundtripDateOnly.vb" id="RoundtripDateOnly":::
+
+The preceding example shows an object with `DateOnly` and `TimeOnly` properties being round-tripped serialized.
+
```
Comment (tdykstra) [inline docs/standard/serialization/system-text-json/how-to.md:99]: I think you can omit this line, but if you want to keep it I'd reword "round-tripped serialized", possibly just dropping "serialized" since that will be obvious in the context of this article.

```suggestion
```

File: docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/csharp/RoundtripDateOnly.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        bool valuesAreTheSame = originalData == deserializedData;
+        Console.WriteLine($"""
+            Original record is the same as the deserialized record: {valuesAreTheSame}
+            """);
+        // </RoundtripDateOnly>
+    }
+}
+
+// <SeriesDataPoint>
+sealed file record class SeriesDataPoint(
```
Comment (tdykstra) [inline docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/csharp/RoundtripDateOnly.cs:36]: I'm not fond of the optional `class` keyword with `record`, as the word `class` implies to me that we'll be getting reference equality. Could make this just a `record` or a `record struct` to emphasize value equality. Just a thought, feel free to ignore.

File: docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/csharp/RoundtripDateOnly.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        string serialized = JsonSerializer.Serialize(originalData);
+
+        Console.WriteLine($"Resulting JSON: {serialized}");
+
+        SeriesDataPoint deserializedData =
+            JsonSerializer.Deserialize<SeriesDataPoint>(serialized)!;
+
+        bool valuesAreTheSame = originalData == deserializedData;
+        Console.WriteLine($"""
+            Original record is the same as the deserialized record: {valuesAreTheSame}
```
Comment (tdykstra) [inline docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/csharp/RoundtripDateOnly.cs:29]:
```suggestion
            Original record has the same values as the deserialized record: {valuesAreTheSame}
```

**Review by tdykstra — APPROVED** [review]
Looks great, I had a couple of suggestions.

File: docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/csharp/RoundtripDateOnly.cs (hunk 2 above)
Comment (IEvangelist) [inline docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/csharp/RoundtripDateOnly.cs:36]:
```suggestion
sealed file record SeriesDataPoint(
```

File: docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/vb/RoundtripDateOnly.vb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+            Dim deserializedData As SeriesDataPoint =
+                JsonSerializer.Deserialize(Of SeriesDataPoint)(serialized)
+            Dim valuesAreTheSame As Boolean = (originalData.DateValue = deserializedData.DateValue AndAlso
+                originalData.Time = deserializedData.Time AndAlso
+                originalData.Id = deserializedData.Id AndAlso
+                originalData.Value = deserializedData.Value)
+
+            Console.WriteLine(
+                $"Original record is the same as the deserialized record: {valuesAreTheSame}")
```
Comment (IEvangelist) [inline docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/vb/RoundtripDateOnly.vb:28]:
```suggestion
                $"Original object has the same values as the deserialized object: {valuesAreTheSame}")
```

File: docs/standard/serialization/system-text-json/how-to.md [hunk 5]
```diff
@@ -85,6 +85,16 @@ The preceding examples use type inference for the type being serialized. An over
...
 :::code language="vb" source="snippets/system-text-json-how-to/vb/RoundtripToString.vb" id="SerializeWithGenericParameter":::
 
+With .NET 7, `System.Text.Json` supports serializing and deserializing <xref:System.DateOnly> and <xref:System.TimeOnly> types. Consider the following object:
+
+:::code source="snippets/system-text-json-how-to/cs/RoundtripDateOnly.cs" id="SeriesDataPoint":::
+:::code source="snippets/system-text-json-how-to/vb/RoundtripDateOnly.vb" id="SeriesDataPoint":::
+
+The following example serializes the `SeriesDataPoint` object, displays the resulting JSON, and then deserializes it back into a new instance of the `SeriesDataPoint` type. Finally, the original and newly deserialized instances are compared for equality and the results are written to the console:
+
+:::code source="snippets/system-text-json-how-to/cs/RoundtripDateOnly.cs" id="RoundtripDateOnly":::
```
Comment (IEvangelist) [inline docs/standard/serialization/system-text-json/how-to.md:95]:
```suggestion
:::code source="snippets/system-text-json-how-to/csharp/RoundtripDateOnly.cs" id="RoundtripDateOnly":::
```

File: docs/standard/serialization/system-text-json/how-to.md [hunk 6]
```diff
@@ -85,6 +85,16 @@ The preceding examples use type inference for the type being serialized. An over
 :::code language="csharp" source="snippets/system-text-json-how-to/csharp/SerializeWithGenericParameter.cs" highlight="23":::
 :::code language="vb" source="snippets/system-text-json-how-to/vb/RoundtripToString.vb" id="SerializeWithGenericParameter":::
 
+With .NET 7, `System.Text.Json` supports serializing and deserializing <xref:System.DateOnly> and <xref:System.TimeOnly> types. Consider the following object:
+
+:::code source="snippets/system-text-json-how-to/cs/RoundtripDateOnly.cs" id="SeriesDataPoint":::
```
Comment (IEvangelist) [inline docs/standard/serialization/system-text-json/how-to.md:90]:
```suggestion
:::code source="snippets/system-text-json-how-to/csharp/RoundtripDateOnly.cs" id="SeriesDataPoint":::
```
